addons/im_livechat/models/discuss_channel_member.py

```diff
@@ -71,16 +73,39 @@ def _compute_agent_expertise_ids(self):
     def _create_or_update_history(self, values_by_member=None):
         if not values_by_member:
             values_by_member = {}
-        member_without_history = self.filtered(lambda m: not m.livechat_member_history_ids)
+        members_without_history = self.filtered(lambda m: not m.livechat_member_history_ids)
+        history_domain = Domain(expression.FALSE_DOMAIN)
+        for member in members_without_history:
+            history_domain |= Domain(
+                [
+                    ("channel_id", "=", member.channel_id.id),
+                    ("partner_id", "=", member.partner_id.id)
+                    if member.partner_id
+                    else ("guest_id", "=", member.guest_id.id),
+                ]
+            )
```

Comment:
Or with just Domain.OR now that I think about it, it's made specifically for this purpose (the idea of the reduce but built-in)

Reply:
Much better with or + list compr indeed.